client/src/pages/admin/MergeLocations.js

```diff
@@ -0,0 +1,516 @@
+import { Button, Callout, Tooltip } from "@blueprintjs/core"
+import { DEFAULT_PAGE_PROPS, DEFAULT_SEARCH_PROPS } from "actions"
+import API from "api"
+import { gql } from "apollo-boost"
+import { LocationOverlayRow } from "components/advancedSelectWidget/AdvancedSelectOverlayRow"
+import AdvancedSingleSelect from "components/advancedSelectWidget/AdvancedSingleSelect"
+import ApprovalSteps, {
+  PLANNING_APPROVAL,
+  PUBLICATION_APPROVAL
+} from "components/approvals/ApprovalSteps"
+import Leaflet from "components/Leaflet"
+import Messages from "components/Messages"
+import { GRAPHQL_NOTES_FIELDS } from "components/Model"
+import {
+  jumpToTop,
+  mapPageDispatchersToProps,
+  PageDispatchersPropType,
+  useBoilerplate
+} from "components/Page"
+import _escape from "lodash/escape"
+import { Location } from "models"
+import GeoLocation from "pages/locations/GeoLocation"
+import PropTypes from "prop-types"
+import React, { useMemo, useState } from "react"
+import { connect } from "react-redux"
+import { useHistory } from "react-router-dom"
+import { toast } from "react-toastify"
+import LOCATIONS_ICON from "resources/locations.png"
+
+const locationFilters = {
+  allLocations: {
+    label: "All locations",
+    queryVars: {}
+  }
+}
+
+const GQL_MERGE_LOCATION = gql`
+  mutation(
+    $loserLocationUuid: String!
+    $loserLocationUuid2: String!
+    $location: LocationInput!
+  ) {
+    mergeLocation(
+      loserLocationUuid: $loserLocationUuid
+      loserLocationUuid2: $loserLocationUuid2
+      location: $location
+    ) {
+      uuid
+    }
+  }
+`
+
+const selectFields = `
+  uuid
+  name
+  lat
+  lng
+  status
+  planningApprovalSteps {
+    uuid
+    name
+    approvers {
+      uuid
+      name
+      person {
+        uuid
+        name
+        rank
+        role
+        avatar(size: 32)
+      }
+    }
+  }
+  approvalSteps {
+    uuid
+    name
+    approvers {
+      uuid
+      name
+      person {
+        uuid
+        name
+        rank
+        role
+        avatar(size: 32)
+      }
+    }
+  }
+  ${GRAPHQL_NOTES_FIELDS}
+`
+
+const LocationColumn = ({
```

Comment:
This seems similar to what a `locations\Form.js` does. Can we adapt it and reuse it?

Reply:
This might be similar to what location Form does but it is fundamentally away from the notion of a form. Particularly, there are no inputs, selects etc. we are not taking any inputs from user. Also we are not performing any validations because we are not getting any input. Adapting location form here wouldn't add any value here. Moreover, it would bring extra complexity to the form. I suggest keeping this isolated as it is now.